kornia/augmentation/augmentation.py

```diff
@@ -1072,28 +1072,47 @@ def inverse(
             padding_size = [0, 0, 0, 0]
         return self.precrop_padding(out, padding_size)
 
+    def forward_parameters_precrop(self, batch_shape) -> Dict[str, torch.Tensor]:
+        input_pad = self.compute_padding(batch_shape)
+        batch_shape = (
+            *batch_shape[:2],
+            batch_shape[2] + input_pad[1] + input_pad[3],  # original height + top + bottom padding
+            batch_shape[3] + input_pad[0] + input_pad[2]  # original width + left + right padding
+        )
+        padding_size = torch.tensor(tuple(input_pad), dtype=torch.long).expand(batch_shape[0], -1)
+        _params = super().forward_parameters(batch_shape)
+        _params.update({"padding_size": padding_size})
+        return _params
+
     def forward(
         self,
         input: TensorWithTransformMat,
         params: Optional[Dict[str, torch.Tensor]] = None,
         return_transform: Optional[bool] = None,
     ) -> TensorWithTransformMat:
+        padding_size = params.get("padding_size") if params else None
+        if padding_size is not None:
+            input_pad = padding_size.unique(dim=0).cpu().squeeze().numpy().tolist()
+        else:
+            input_pad = None
```

Comment:
```suggestion
        input_pad = params.get("padding_size").unique(dim=0).cpu().squeeze().numpy().tolist() if params else None
```

Reply:
This assumes that whenever `params` is not None `padding_size` will be present, which I guess is safe to assume now since either you set the `padding_size` when running `forward_parameters_precrop` when inside a container or you set it after doing the crop when not. 

This however reminded me I have not touched the other two containers `PatchSequential` nor `VideoSequential`, and those are probably broken too. I'll take a look.

Reply:
> This assumes that whenever `params` is not None `padding_size` will be present, which I guess is safe to assume now since either you set the `padding_size` when running `forward_parameters_precrop` when inside a container or you set it after doing the crop when not.

The current param generation pipeline will always have this parameter. Right?

> This however reminded me I have not touched the other two containers `PatchSequential` nor `VideoSequential`, and those are probably broken too. I'll take a look.

I think the principle here is to make it work without touching those containers. At least, we shall not make one augmentation special to others. Better keep it generic.

Reply:
> The current param generation pipeline will always have this parameter. Right?


The current param generation pipeline did not have this parameter before changes added here and that is still the case (just checked) with `VideoSequential`, which has the same problem as the containers had and is not fixed by the changes to `ImageSequential.forward_parameters` since it implements its own.

Therefore, the change you propose here makes tests fail in VideoSequential when adding a RandomCrop with padding into the augmentations since the padding_size parameter is missing. But the crop is incorrectly done anyway so.. :)

> I think the principle here is to make it work without touching those containers. At least, we shall not make one augmentation special to others. Better keep it generic.

Ideally yes, but since RandomCrop is indeed a special case, it will require more changes to avoid doing so I feel. I would suggest to patch it like this (also for `VideoSequential`) for now so behaviour for RandomCrop is consistent inside and out side of containers and correct in both. And then explore what you mentioned about a `preprocessing` hook so we can remove this ugly patch later, or just make sure we always generate the parameters for RandomCrop before the crop is done so we do not have to deal with is as a special case.